src/main/java/com/google/sps/managers/DatastoreManager.java

```diff
@@ -0,0 +1,327 @@
+    userEntity.setProperty("profilePictureUrl", user.getProfilePictureUrl());
+    String numbersJson = gson.toJson(user.getGamesCompletedWithTime());
+    userEntity.setProperty("gamesCompletedWithTime", numbersJson);
+    datastore.put(userEntity);
+  }
+  
+  /**
+    * Retrieves a single user entity from the datastore.
+    * @param userID the unique code used to identify this specific user.
+    * @return a User object with the properties specified in the Builder.
+  */
+  public User retrieveUser(String userID) {
+    Key userEntityKey = KeyFactory.createKey("User", userID);
+    Entity userEntity;
+    try {
+      userEntity = datastore.get(userEntityKey);
+    }
+    catch(Exception e) {
+      return null;
+    }
+    String userName = (String) userEntity.getProperty("username");
+    String firstName = (String) userEntity.getProperty("firstname");
+    String lastName = (String) userEntity.getProperty("lastname");
+    ArrayList<String> gamesCreated = (ArrayList<String>) userEntity.getProperty("gamesCreated");
+    String profilePictureUrl = (String) userEntity.getProperty("profilePictureUrl");
+    String gamesCompletedWithTimeJson = (String) userEntity.getProperty("gamesCompletedWithTime");
+    Type pairListType = new TypeToken<ArrayList<Pair<String, Long>>>(){}.getType();
+    ArrayList<Pair<String, Long>> gamesCompletedWithTime = gson.fromJson(gamesCompletedWithTimeJson, pairListType);
+  
+    User.Builder user = new User.Builder(userID).setUsername(userName).setFirstName(firstName).setLastName(lastName);
+    user.setProfilePictureUrl(profilePictureUrl).setGamesCreated(gamesCreated);
+    user.setGamesCompletedWithTime(gamesCompletedWithTime);
+    return user.build();
+  }
+
+  /**
+    * Creates or Replaces the static data of a single game in datastore as an entity
+    * @param game an Game variable representing a single instance of a game.
+  */
+  public void createOrReplaceGame(Game game) {
+    Entity gameEntity = new Entity("Game", game.getGameID());
+    gameEntity.setProperty("gameID", game.getGameID());
+    gameEntity.setProperty("gameName", game.getGameName());
+    gameEntity.setProperty("gameCreator", game.getGameCreator());
+    gameEntity.setProperty("gameDescription", game.getGameDescription());
+    gameEntity.setProperty("stages", game.getStages());
+    gameEntity.setProperty("numTimesPlayed", game.getNumTimesPlayed());
+    gameEntity.setProperty("numTimesFinished", game.getNumTimesFinished());
+    gameEntity.setProperty("numStarVotes", game.getNumStarVotes());
+    gameEntity.setProperty("totalStars", game.getTotalStars());
+    gameEntity.setProperty("numDifficultyVotes", game.getNumDifficultyVotes());
+    gameEntity.setProperty("totalDifficulty", game.getTotalDifficulty());
+    datastore.put(gameEntity);
+  }
+  
+  /**
+    * Retrieves a single game entity from the datastore.
+    * @param gameID the unique code used to identify this specific game.
```

Comment:
@param gameID is the unique code used to identify which specific game to be retrieved